config/dev/vsphere-credentials.yaml

```diff
@@ -24,10 +28,76 @@ kind: Credential
 metadata:
   name: vsphere-cluster-identity-cred
   namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
 spec:
   description: vSphere credentials
   identityRef:
     apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
     kind: VSphereClusterIdentity
     name: vsphere-cluster-identity
     namespace: ${NAMESPACE}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vsphere-cluster-identity-resource-template
+  namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
+  annotations:
+    projectsveltos.io/template: "true"
+data:
+  configmap.yaml: |
+    {{- $$cluster := .InfrastructureProvider -}}
+    {{- $$identity := (getResource "InfrastructureProviderIdentity") -}}
+    {{- $$secret := (getResource "InfrastructureProviderIdentitySecret") -}}
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: vsphere-cloud-secret
+      namespace: kube-system
+    type: Opaque
+    data:
+      {{ printf "%s.username" $$cluster.spec.server }}: {{ index $$secret.data "username" }}
+      {{ printf "%s.password" $$cluster.spec.server }}: {{ index $$secret.data "password" }}
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: vcenter-config-secret
+      namespace: kube-system
+    type: Opaque
+    stringData:
+      csi-vsphere.conf: |
+        [Global]
+        cluster-id = "{{ $$cluster.metadata.name }}"
+
+        [VirtualCenter "{{ $$cluster.spec.server }}"]
+        insecure-flag = "true"
+        user = "{{ index $$secret.data "username" | b64dec }}"
+        password = "{{ index $$secret.data "password" | b64dec }}"
+        port = "443"
+        datacenters = ${VSPHERE_DATACENTER}
+    ---
+    apiVersion: v1
+    kind: ConfigMap
+    metadata:
```

Comment:
the component label is missing here. i am not sure whether sveltos will properly recreate everything and in the correct moment that's why i'd put the label here to avoid potential inconsistencies